epowcore/pandapower/pandapower_model.py

```diff
@@ -0,0 +1,439 @@
+            tap_step_percent=transformer3w.TapDetails.tap_changer_voltage * 100,
+            tap_step_degree=nan,
+            tap_pos=transformer3w.TapDetails.tap_initial,
+            tap_neutral=transformer3w.TapDetails.tap_neutral,
+            tap_max=transformer3w.TapDetails.tap_max,
+            tap_min=transformer3w.TapDetails.tap_min,
+            in_service=True,
+            max_loading_percent=nan,
+            tap_at_star_point=False,
+            vk0_hv_percent=nan,
+            vk0_mv_percent=nan,
+            vk0_lv_percent=nan,
+            vkr0_hv_percent=nan,
+            vkr0_mv_percent=nan,
+            vkr0_lv_percent=nan,
+            vector_group=None,
+            tap_dependent_impedance=nan,
+            vk_hv_percent_characteristic=None,
+            vkr_hv_percent_characteristic=None,
+            vk_mv_percent_characteristic=None,
+            vkr_mv_percent_characteristic=None,
+            vk_lv_percent_characteristic=None,
+            vkr_lv_percent_characteristic=None,
+        )
+        return True
+
+    def create_generator_from_gdf_synchronous_machine(
+        self, core_model: CoreModel, synchronous_machine: SynchronousMachine
+    ):
+        """Create a generator in the pandapower network equivalent to
+        the given synchronous_machine in gdf format.
+        """
+        # Getting the bus the generator is connected to
+        synchronous_machine_bus = get_connected_bus(
+            core_model.graph, synchronous_machine, max_depth=1
+        )
+        # If the bus wasnt found the function fails
+        if synchronous_machine_bus is None:
+            Logger.log_to_selected("Failed to convert synchonous_machine")
+            return False
+
+        # Check if the generator bus was slack
+        slack = False
+        if synchronous_machine_bus.lf_bus_type == LFBusType("SLACK"):
+            slack = True
+            Logger.log_to_selected(
+                "Gen:" + synchronous_machine.name + " is set to be a slack genenerator"
+            )
+        # Create generator in pandapower network
+        pandapower.create_gen(
+            net=self.network,
+            name=synchronous_machine.name,
+            index=synchronous_machine.uid,
+            bus=synchronous_machine_bus.uid,
+            p_mw=synchronous_machine.active_power,
+            vm_pu=synchronous_machine.voltage_set_point,
+            sn_mva=synchronous_machine.rated_apparent_power,
+            max_q_mvar=synchronous_machine.q_max,
+            min_q_mvar=synchronous_machine.q_min,
+            min_p_mw=synchronous_machine.p_min,
+            max_p_mw=synchronous_machine.p_max,
+            min_vm_pu=nan,
+            max_vm_pu=nan,
+            scaling=1.0,
+            type="sync",
+            slack=slack,
+            controllable=False,
+            vn_kv=synchronous_machine.rated_voltage,
+            xdss_pu=synchronous_machine.subtransient_reactance_x,
+            rdss_ohm=nan,
+            cos_phi=nan,
```

Comment:
This should be `cosn` in PowerFactory. We can calculate this as `cosn = rated_active_power / rated_apparent_power`. This could be added to the `SynchronousMachine` class as a property.